pydantic/schema.py

```diff
@@ -568,15 +574,17 @@ def model_type_schema(
     properties = {}
     required = []
     definitions: Dict[str, Any] = {}
+    nested_models: Set[str] = set()
     for k, f in model.__fields__.items():
         try:
-            f_schema, f_definitions = field_schema(
+            f_schema, f_definitions, f_nested_models = field_schema(
                 f, by_alias=by_alias, model_name_map=model_name_map, ref_prefix=ref_prefix, known_models=known_models
             )
         except SkipField as skip:
             warnings.warn(skip.message, UserWarning)
             continue
         definitions.update(f_definitions)
+        nested_models |= f_nested_models
```

Comment:
Since this is most likely all an upfront cost I don't believe it's a big deal but you can probably update in-place
```suggestion
        nested_models.update(f_nested_models)
```